tests/Integration/Subscriber/ExpiredCachePurgeSubscriber/TestCleanCacheScheduledEvent.php

```diff
@@ -0,0 +1,80 @@
+<?php
+namespace WP_Rocket\Tests\Integration\Subscriber\ExpiredCachePurgeSubscriber;
+
+use WP_Rocket\Subscriber\Cache\Expired_Cache_Purge_Subscriber;
+use WP_Rocket\Tests\Integration\TestCase;
+use Brain\Monkey\Functions;
+
+/**
+ * @group Subscriber_ScheduledEvent
+ */
+class TestCleanCacheScheduledEvent extends TestCase {
+	public function testShouldNotCleanScheduledEventWhenValuesAreTheSame() {
+		Functions\expect( 'wp_clear_scheduled_hook' )->never();
+
+		update_option(
+			'wp_rocket_settings',
+			[
+				'purge_cron_interval' => 10,
+				'purge_cron_unit'     => 'HOUR_IN_SECONDS',
+			]
+		);
+
+		update_option(
+			'wp_rocket_settings',
+			[
+				'purge_cron_interval' => 10,
+				'purge_cron_unit'     => 'HOUR_IN_SECONDS',
+			]
+		);
+
+		$this->assertTrue( true ); // Prevent "risky" warning.
+	}
+
+	public function testShouldNotCleanScheduledEventWhenChangedValueFromHoursToDays() {
+		Functions\expect( 'wp_clear_scheduled_hook' )->never();
+
+		update_option(
+			'wp_rocket_settings',
+			[
+				'purge_cron_interval' => 10,
+				'purge_cron_unit'     => 'HOUR_IN_SECONDS',
+			]
+		);
+
+		update_option(
+			'wp_rocket_settings',
+			[
+				'purge_cron_interval' => 10,
+				'purge_cron_unit'     => 'DAY_IN_SECONDS',
+			]
+		);
+
+		$this->assertTrue( true ); // Prevent "risky" warning.
+	}
+
+	public function testShouldCleanScheduledEventWhenMinutesAndOldValueIsHours() {
+		Functions\expect( 'wp_clear_scheduled_hook' )
+			->once()
+			->with( Expired_Cache_Purge_Subscriber::EVENT_NAME )
+			->andReturnNull(); // No need to run it.
```

Comment:
@crystinutzaa Just to make sure it's called for this test, I checked the arg with `with`. And to prevent it from running, I added `andReturnNull()`, ie we don't need `wp_clear_scheduled_hook` to run; rather, we just want to know that we made it this far to validate this test works as expected.